lib/razor/matcher.rb

```diff
@@ -306,5 +318,13 @@ def validate(rule, required_returns, caller_name = nil, caller_position = nil)
         end
       end
     end
+
+    if name == 'like' and errors.empty?
+      begin
+        Regexp.new(rule[2])
+      rescue RegexpError => e
+        errors << _("invalid regular expression supplied to `like` for argument %{position}") % {position: 1}
```

Comment:
You might want to include e.message in the text as well?  It might give more context to a user about the specific error.